### ISSUE TITLE
Server boot crash issue with Immersive Portal 0.8-mc1.16.4-forge on Forge-1.16.4-35.1.1

### ISSUE THREAD

**Issue by solarblindfull** [qouteall/ImmersivePortalsModForForge#181]
Hello !

I'm on Forge-1.16.4-35.1.1 with Immersive Portal 0.8-mc1.16.4-forge (the last one with better GUI). I have a server boot crash issue as the lastest server log shows below.

<details>
  <summary>Lastest_log</summary>

  ```
  [22nov.2020 18:18:45.207] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer/]: Starting minecraft server version 1.16.4
[22nov.2020 18:18:45.208] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer/]: Loading properties
[22nov.2020 18:18:45.212] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer/]: Default game type: SURVIVAL
[22nov.2020 18:18:45.212] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: Generating keypair
[22nov.2020 18:18:45.365] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer/]: Starting Minecraft server on 62.210.51.33:10034
[22nov.2020 18:18:45.418] [Server thread/INFO] [net.minecraft.network.NetworkSystem/]: Using epoll channel type
[22nov.2020 18:18:45.759] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Configuration file ./world_server/serverconfig/the_bumblezone-mod_compatibility.toml is not correct. Correcting
[22nov.2020 18:18:45.766] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Configuration file ./world_server/serverconfig/the_bumblezone-block_mechanics.toml is not correct. Correcting
[22nov.2020 18:18:45.775] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Configuration file ./world_server/serverconfig/the_bumblezone-bee_aggression.toml is not correct. Correcting
[22nov.2020 18:18:45.779] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Configuration file ./world_server/serverconfig/the_bumblezone-dimension.toml is not correct. Correcting
[22nov.2020 18:18:45.782] [Server thread/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Configuration file ./world_server/serverconfig/the_bumblezone-dungeons.toml is not correct. Correcting
[22nov.2020 18:18:45.852] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer/]: Preparing level "world_server"
[22nov.2020 18:18:47.477] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing start
[22nov.2020 18:18:47.486] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing start
[22nov.2020 18:18:47.510] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing start
[22nov.2020 18:18:47.510] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing end
[22nov.2020 18:18:47.514] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing start
[22nov.2020 18:18:47.514] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing end
[22nov.2020 18:18:47.519] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing start
[22nov.2020 18:18:47.519] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing end
[22nov.2020 18:18:47.523] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing start
[22nov.2020 18:18:47.523] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing end
[22nov.2020 18:18:47.527] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing start
[22nov.2020 18:18:47.528] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing end
[22nov.2020 18:18:47.531] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing start
[22nov.2020 18:18:47.532] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing end
[22nov.2020 18:18:47.535] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing start
[22nov.2020 18:18:47.535] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing end
[22nov.2020 18:18:47.538] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing start
[22nov.2020 18:18:47.538] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing end
[22nov.2020 18:18:47.552] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing start
[22nov.2020 18:18:47.553] [Server thread/INFO] [com.nhwhite3118.cobbler.Cobbler/]: addDimensionalSpacing end
[22nov.2020 18:18:47.861] [Server thread/INFO] [Portal/]: Successfully read IP's dimension id record
[22nov.2020 18:18:47.862] [Server thread/INFO] [Portal/]: Start Completing Dimension Id Record
[22nov.2020 18:18:47.863] [Server thread/INFO] [Portal/]: Before:
minecraft:overworld -> 0
immersive_portals:alternate1 -> 2
world_blender:world_blender -> 9
immersive_portals:alternate2 -> 3
the_bumblezone:the_bumblezone -> 7
minecraft:the_nether -> -1
immersive_portals:alternate3 -> 4
undergarden:undergarden -> 8
immersive_portals:alternate4 -> 5
minecraft:the_end -> 1
immersive_portals:alternate5 -> 6
[22nov.2020 18:18:47.864] [Server thread/INFO] [Portal/]: Server Loaded Dimensions:
immersive_portals:alternate1
immersive_portals:alternate2
immersive_portals:alternate3
immersive_portals:alternate4
immersive_portals:alternate5
minecraft:overworld
minecraft:the_end
minecraft:the_nether
the_bumblezone:the_bumblezone
undergarden:undergarden
world_blender:world_blender

[22nov.2020 18:18:47.866] [Server thread/INFO] [Portal/]: After:
minecraft:overworld -> 0
immersive_portals:alternate1 -> 2
world_blender:world_blender -> 9
immersive_portals:alternate2 -> 3
the_bumblezone:the_bumblezone -> 7
minecraft:the_nether -> -1
immersive_portals:alternate3 -> 4
undergarden:undergarden -> 8
immersive_portals:alternate4 -> 5
minecraft:the_end -> 1
immersive_portals:alternate5 -> 6
[22nov.2020 18:18:47.867] [Server thread/INFO] [Portal/]: Dimension Id Info Saved to File
[22nov.2020 18:18:47.874] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: Preparing start region for dimension minecraft:overworld
[22nov.2020 18:18:50.014] [Server thread/INFO] [net.minecraft.world.chunk.listener.LoggingChunkStatusListener/]: Preparing spawn area: 0%
[22nov.2020 18:18:50.015] [Server thread/INFO] [net.minecraft.world.chunk.listener.LoggingChunkStatusListener/]: Preparing spawn area: 0%
[22nov.2020 18:18:50.015] [Server thread/INFO] [net.minecraft.world.chunk.listener.LoggingChunkStatusListener/]: Preparing spawn area: 0%
[22nov.2020 18:18:50.016] [Server thread/INFO] [net.minecraft.world.chunk.listener.LoggingChunkStatusListener/]: Preparing spawn area: 0%
[22nov.2020 18:18:50.016] [Server thread/INFO] [net.minecraft.world.chunk.listener.LoggingChunkStatusListener/]: Preparing spawn area: 0%
[22nov.2020 18:18:51.042] [Server thread/INFO] [net.minecraft.world.chunk.listener.LoggingChunkStatusListener/]: Preparing spawn area: 0%
[22nov.2020 18:18:51.042] [Server thread/INFO] [net.minecraft.world.chunk.listener.LoggingChunkStatusListener/]: Preparing spawn area: 0%
[22nov.2020 18:18:51.979] [Server thread/INFO] [net.minecraft.world.chunk.listener.LoggingChunkStatusListener/]: Time elapsed: 4104 ms
[22nov.2020 18:18:51.982] [Server thread/INFO] [Portal/]: Loading custom portal gen
[22nov.2020 18:18:53.905] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer/]: Done (8.186s)! For help, type "help"
[22nov.2020 18:18:53.906] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer/]: Starting GS4 status listener
[22nov.2020 18:18:53.915] [Server thread/INFO] [net.minecraft.network.rcon.RConThread/]: Thread Query Listener started
[22nov.2020 18:18:53.915] [Query Listener #1/INFO] [net.minecraft.network.rcon.QueryThread/]: Query running on 62.210.51.33:10034
[22nov.2020 18:18:53.950] [Server thread/INFO] [blueduck.morejellyfish.morejellyfishmod.MoreJellyfishMod/]: HELLO from server starting
[22nov.2020 18:18:53.950] [Server thread/INFO] [blueduck.jellyfishing.jellyfishingmod.JellyfishingMod/]: HELLO from server starting
[22nov.2020 18:18:53.954] [Server thread/INFO] [corgiaoc.byg.BYG/]: BYG: "Server Starting" Event Complete!
[22nov.2020 18:18:54.609] [Server thread/FATAL] [net.minecraftforge.fml.loading.RuntimeDistCleaner/DISTXFORM]: Attempted to load class net/minecraft/client/world/ClientWorld for invalid dist DEDICATED_SERVER
[22nov.2020 18:18:54.610] [Server thread/ERROR] [net.minecraft.server.MinecraftServer/]: Encountered an unexpected exception
java.lang.RuntimeException: Attempted to load class net/minecraft/client/world/ClientWorld for invalid dist DEDICATED_SERVER
	at net.minecraftforge.fml.loading.RuntimeDistCleaner.processClassWithFlags(RuntimeDistCleaner.java:71) ~[forge-1.16.4-35.1.1.jar:35.1]
	at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:85) ~[modlauncher-8.0.6.jar:?]
	at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:119) ~[modlauncher-8.0.6.jar:?]
	at cpw.mods.modlauncher.TransformingClassLoader$DelegatedClassLoader.findClass(TransformingClassLoader.java:265) ~[modlauncher-8.0.6.jar:?]
	at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:136) ~[modlauncher-8.0.6.jar:?]
	at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:98) ~[modlauncher-8.0.6.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[?:1.8.0_202]
	at com.qouteall.immersive_portals.chunk_loading.EntitySync.lambda$tick$1(EntitySync.java:62) ~[immersive_portals:0.8]
	at java.util.LinkedHashMap$LinkedValues.forEach(LinkedHashMap.java:608) ~[?:1.8.0_202]
	at com.qouteall.immersive_portals.chunk_loading.EntitySync.tick(EntitySync.java:57) ~[immersive_portals:0.8]
	at com.qouteall.immersive_portals.my_util.Signal.lambda$emit$0(Signal.java:18) ~[immersive_portals:0.8]
	at java.util.ArrayList.forEach(ArrayList.java:1257) ~[?:1.8.0_202]
	at com.qouteall.immersive_portals.my_util.Signal.emit(Signal.java:18) ~[immersive_portals:0.8]
	at net.minecraft.server.MinecraftServer.handler$zef000$onServerTick(MinecraftServer.java:2741) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:878) ~[?:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:291) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:786) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:641) ~[?:?]
	at net.minecraft.server.MinecraftServer.lambda$startServer$0(MinecraftServer.java:232) ~[?:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_202]
[22nov.2020 18:18:54.627] [Server thread/FATAL] [net.minecraftforge.common.ForgeMod/]: Preparing crash report with UUID f9a78c62-d772-47ca-8809-bcad813fda55
  ```
</details>

The issue occurs at the boot of my server with a manually dimension stacked world. To make my custom dimension stack world I typed the commands below in singleplayer mode.

<details>
  <summary>Command_lines</summary>

  ```
/gamerule ipDimensionStack true

/execute in undergarden:undergarden run teleport [player] 0 10 0

/portal global connect_floor undergarden:undergarden minecraft:the_nether

/portal global convert_global_portal_to_normal_portal

/portal move_portal_destination 128

/portal rotate_portal_rotation_along x 180

/portal global convert_normal_portal_to_global_portal

------------------------------------------------------------------------------------------------

/execute in minecraft:the_nether run teleport [player] 0 10 0

/portal global connect_floor minecraft:the_nether undergarden:undergarden

/portal global convert_global_portal_to_normal_portal

/portal move_portal_destination 256

/portal rotate_portal_rotation_along x 180

/portal move_portal_destination 2

/portal global convert_normal_portal_to_global_portal

------------------------------------------------------------------------------------------------

/portal global connect_floor minecraft:overworld undergarden:undergarden

/portal global connect_ceil undergarden:undergarden minecraft:overworld

/portal global connect_floor world_blender:world_blender minecraft:the_nether

/portal global connect_ceil minecraft:the_nether world_blender:world_blender
  ```
</details>

This world config works perfectly without a lag in singleplayer. Can you see what's wrong please ?
Thank you very much

Have a good day !

**Comment by qouteall** [comment]
maybe fixed in 0.9

**Comment by solarblindfull** [comment]
Thank you for your quick reply.
I tested your fix and all works very well ! Thank you very much ! :D

Have a good day
